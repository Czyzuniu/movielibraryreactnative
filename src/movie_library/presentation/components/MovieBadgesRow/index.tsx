import {Box, HStack, IImageProps, useTheme} from 'native-base';
import React from 'react';
import MovieInfoBadge from '../MovieInfoBadge';
import CoreMapper from '../../../../base/mapper/CoreMapper';
import dayjs from 'dayjs';
import Icon from 'react-native-vector-icons/FontAwesome';
import DescriptiveMovie from '../../../domain/entity/DescriptiveMovie';
import {useTranslation} from 'react-i18next';

type Props = IImageProps & {
  movie: DescriptiveMovie;
};

export default function MovieBadgesRow({movie}: Props) {
  const {colors} = useTheme();
  const {t} = useTranslation('viewMovie');

  return (
    <Box m={1.5}>
    <HStack margin={1.5} space={5} flexDirection={'row'}>
      <MovieInfoBadge
        mainText={CoreMapper.numberToString(movie.averageVote)}
        secondaryText={t('TotalVotes', {
          total: CoreMapper.numberToString(movie.totalVotes),
        })}
        icon={<Icon name={'star'} color={colors.amber['600']}/>}
      />
      <MovieInfoBadge
        mainText={t('RuntimeMin', {minutes: movie.runTime})}
        secondaryText={t('Runtime')}
        icon={<Icon name={'clock-o'} color={colors.amber['600']}/>}
      />
      <MovieInfoBadge
        mainText={dayjs(movie.releaseDate).format('MMM, D YYYY')}
        secondaryText={t('ReleaseDate')}
        icon={<Icon name={'calendar'} color={colors.amber['600']}/>}
      />
    </HStack>
    </Box>
  );
}
