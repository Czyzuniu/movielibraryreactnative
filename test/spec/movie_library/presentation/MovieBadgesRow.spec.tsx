import {render} from '@testing-library/react-native';
import React from 'react';
import ContextWrapper from '../../../utils/ContextWrapper';
import MovieBadgesRow from '../../../../src/movie_library/presentation/components/MovieBadgesRow';
import TestObjects from '../../../utils/TestObjects';

describe('MovieBadgesRow Tests', () => {
  it('should render and match snapshot', () => {
    const {toJSON} = render(
      <ContextWrapper>
        <MovieBadgesRow movie={TestObjects.DescriptiveMovie()} />
      </ContextWrapper>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
