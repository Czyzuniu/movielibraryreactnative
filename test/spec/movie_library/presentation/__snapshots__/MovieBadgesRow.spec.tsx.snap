// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieBadgesRow Tests should render and match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "flex": 1,
        "height": 90,
        "marginBottom": 6,
        "marginLeft": 6,
        "marginRight": 6,
        "marginTop": 6,
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        dataSet={Object {}}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#fafafa",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 21,
                "marginBottom": 4,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 4,
                "textDecorationLine": undefined,
              }
            }
          >
            8
          </Text>
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#d97706",
                  "fontSize": 12,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          dataSet={Object {}}
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#fafafa",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          TotalVotes
        </Text>
      </View>
      <View
        dataSet={Object {}}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#fafafa",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 21,
                "marginBottom": 4,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 4,
                "textDecorationLine": undefined,
              }
            }
          >
            RuntimeMin
          </Text>
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#d97706",
                  "fontSize": 12,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          dataSet={Object {}}
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#fafafa",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          Runtime
        </Text>
      </View>
      <View
        dataSet={Object {}}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#fafafa",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 21,
                "marginBottom": 4,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 4,
                "textDecorationLine": undefined,
              }
            }
          >
            Apr, 24 2022
          </Text>
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#d97706",
                  "fontSize": 12,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          dataSet={Object {}}
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#fafafa",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          ReleaseDate
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
